### PR TITLE
BAEL-2477

### PR DESCRIPTION
**Pull request by myluckagain**

### CODE REVIEW

File: persistence-modules/spring-data-jpa/src/test/java/com/baeldung/customer/CustomerRepositoryIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+@DataJpaTest(excludeAutoConfiguration = {
+        PersistenceConfiguration.class,
+        PersistenceUserConfiguration.class,
+        PersistenceProductConfiguration.class})
+@RunWith(SpringRunner.class)
+public class CustomerRepositoryIntegrationTest {
+
+    @PersistenceContext
+    private EntityManager entityManager;
+    @Autowired
```
Comment (ashleyfrieze) [inline persistence-modules/spring-data-jpa/src/test/java/com/baeldung/customer/CustomerRepositoryIntegrationTest.java]: A space between these properties might be helpful

File: persistence-modules/spring-data-jpa/src/test/java/com/baeldung/customer/CustomerRepositoryIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    public void before() {
+        entityManager.persist(new Customer("A", "A@example.com"));
+        entityManager.persist(new Customer("D", null));
+        entityManager.persist(new Customer("D", "D@example.com"));
+    }
+
+    @Test
+    public void givenQueryMethod_whenEmailIsNull_thenFoundByNullEmail() {
+        List<Customer> customers = repository.findByNameAndEmail("D", null);
+        assertEquals(1, customers.size());
```
Comment (KevinGilmore) [inline persistence-modules/spring-data-jpa/src/test/java/com/baeldung/customer/CustomerRepositoryIntegrationTest.java]: Let's add a blank line before the assertions begin. This separates the "when" and "then" parts of the test.

File: persistence-modules/spring-data-jpa/src/test/java/com/baeldung/customer/CustomerRepositoryIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        Customer actual = customers.get(0);
+        assertEquals(null, actual.getEmail());
+        assertEquals("D", actual.getName());
+    }
+
+    @Test
+    public void givenQueryMethod_whenEmailIsAbsent_thenIgnoreEmail() {
+
```
Comment (KevinGilmore) [inline persistence-modules/spring-data-jpa/src/test/java/com/baeldung/customer/CustomerRepositoryIntegrationTest.java]: remove blank line

File: persistence-modules/spring-data-jpa/src/test/java/com/baeldung/customer/CustomerRepositoryIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        List<Customer> customers = repository.findByName("D");
+        assertEquals(2, customers.size());
```
Comment (KevinGilmore) [inline persistence-modules/spring-data-jpa/src/test/java/com/baeldung/customer/CustomerRepositoryIntegrationTest.java]: add blank line before assertion

File: persistence-modules/spring-data-jpa/src/test/java/com/baeldung/customer/CustomerRepositoryIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    @Test
+    public void givenQueryAnnotation_whenEmailIsNull_thenIgnoreEmail() {
+        List<Customer> customers = repository.findCustomerByNameAndEmail("D", null);
+        assertEquals(2, customers.size());
```
Comment (KevinGilmore) [inline persistence-modules/spring-data-jpa/src/test/java/com/baeldung/customer/CustomerRepositoryIntegrationTest.java]: add blank line before assertion

File: spring-boot-mvc/pom.xml [hunk 6]
```diff
@@ -72,6 +72,16 @@
             <version>${spring.fox.version}</version>
         </dependency>
 
+        <dependency>
```
Comment (KevinGilmore) [inline spring-boot-mvc/pom.xml]: Why is anything from the spring-boot-mvc module included in this PR?

File: persistence-modules/spring-data-jpa/src/main/java/com/baeldung/customer/CustomerRepository.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+import java.util.List;
+
+public interface CustomerRepository extends JpaRepository<Customer, Long> {
+
+    List<Customer> findByName(String name);
+
+    List<Customer> findByNameAndEmail(String name, String email);
+
+    @Query("SELECT s FROM Customer s WHERE (:name is null or s.name = :name) and (:email is null or s.email = :email)")
```
Comment (KevinGilmore) [inline persistence-modules/spring-data-jpa/src/main/java/com/baeldung/customer/CustomerRepository.java]:
```suggestion
    @Query("SELECT c FROM Customer c WHERE (:name is null or c.name = :name) and (:email is null or c.email = :email)")
```

**Pull request closed without merging**